### PR TITLE
(inkscape) Create start menu entries for all users

### LINKED ISSUE #2328: (inkscape)  Start menu entry is not visible for all users
### Chocolatey Version

2.2.2

### Chocolatey License

None

### Package Version

1.3.0

### Current Behaviour

Start Menu entry is created under %APPDATA% folder of the user who runs Choco
--> Start menu entry is not visible to other users

### Expected Behaviour

Start menu entry is visible to other users

### Steps To Reproduce

- log in as Administrator user
- in administrative powershell: `choco install inkscape`
- switch to other user
- --> Inkscape not in start menu

### Environment

```markdown
- Operating System: Microsoft Windows NT 10.0.20348.0  (Win Server 2022 demo VM from Microsoft)
- PowerShell Version: 5.1.20348.1850
- Shell: Power Shell, right click - run as Administrator
```

### Chocolatey Log

```shell
not relevant
```

### Anything else?

Upstream bug: https://gitlab.com/inkscape/inkscape/-/issues/1618
Workaround: Install MSI package with extra argument `ALLUSERS=1`.
Example: `msiexec /qn /i inkscape-XXXX.msi ALLUSERS=1` works correctly.

--> Suggested fix: Append `ALLUSERS=1` to `silentArgs` here:
https://github.com/chocolatey-community/chocolatey-packages/blob/b2226a47b637fc643fdd6f70f27438ac0ecd48bb/automatic/inkscape/tools/chocolateyInstall.ps1#L13

I don't know enough about Choco to test this suggestion.

### PR DESCRIPTION
**Pull request by mgmax**
Make start menu entries visible for all users.
Workaround for upstream bug https://gitlab.com/inkscape/inkscape/-/issues/1618

## Description
Apply the workaround suggested in the upstream bug: Pass ALLUSERS=1 to the MSI installer.
Then, the start menu entries are visible for all users and not just for the one user that runs choco.

## Motivation and Context
Fixes #2328 

## How Has this Been Tested?
- official choco test env 
- and also in custom VM
- You can see that it works by right-clicking on the Inkscape start menu entry --> More --> Open File Location. Check that it is NOT in C:\Users\...\AppData\... but somewhere system-wide

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Migrated package (a package has been migrated from another repository)

## Checklist:

- [x] My code follows the code style of this repository.
- [ ] My change requires a change to documentation (this usually means the notes in the description of a package).
- [ ] I have updated the documentation accordingly (this usually means the notes in the description of a package).
- [ ] I have updated the package description and it is less than 4000 characters.
- [x] All files are up to date with the latest [Contributing Guidelines](https://github.com/chocolatey-community/chocolatey-packages/blob/master/CONTRIBUTING.md)
- [x] The added/modified package passed install/uninstall in the chocolatey test environment.
- [x] The changes only affect a single package (not including meta package).

### CODE REVIEW

**Comment by mgmax** [comment]
I tested it locally:
Download the MSI to `...../inkscape/tools` within chocolatey-packages
In Admin-Powershell, run
```
cd ......./inkscape/tools
choco pack
choco install --source .
```

And it works as expected. Is this enough?

**Comment by AdmiringWorm** [comment]
@mgmax Unfortunately, testing on in your own local environment or a custom VM is not enough for changes that affect installs/upgrades or uninstalls.

Any changes that affect those must be tested in the [chocolatey-test-environment](https://github.com/chocolatey-community/chocolatey-test-environment), no matter how insignificant you believe the change is.

The PR will not be merged until it has been tested in this environment, and if you are waiting on us to do that test, that may take weeks, or even months before we have the time available to do such tests.

**Comment by mgmax** [comment]
@AdmiringWorm I tested it successfully with the official test environment.

Additionally, I also tested the case of first installing the current package version and then force-installing my modified version. This also works correctly.

(
As a side note, it was quite bumpy to get the test environment running, mainly due to upstream issues - just to give you an impression:

- more disk space required than stated in the readme (will open a separate PR here)
- VirtualBox Windows WinRM not working out of the box on Debian Linux
- VirtualBox nested virtualization does not work (Crash / "Guru Meditation" when starting VirtualBox within VirtualBox), therefore I really needed a physical Windows host
- weird `vagrant` WinRM errors "The device is not ready" that are actually due to timeouts (will open a separate issue with `vagrant`)
- `vagrant` boot timeout on slow hardware (will open a separate PR here)
- sporadic `vagrant` failures https://github.com/hashicorp/vagrant/issues/9023

)

Happy to get this merged, thanks for your work :-)

**Comment by mgmax** [comment]
The PR should be ready to review. Please let me know if there is anything I can do to get it merged.

**Comment by pauby** [comment]
@mgmax Given [this comment](https://gitlab.com/inkscape/inkscape/-/issues/1618#note_1648843153), it might be worthwhile just holding off on a fix from the author.

I'm concerned that we fix it now, then the author fixes it, then we have to remove our fix (or our fix becomes redundant and confusing for future us's.

**Comment by mgmax** [comment]
Thanks for your reply. I would expect that it takes roughly one year for the fix to land. (CMake release --> MSYS package --> Inkscape build tooling --> next inkscape release)

The workaround does no harm even after the fix has landed. 

Is this one year worth the price of a workaround here? I would say yes, as the current bug makes Inkscape on chocolatey half-broken. You can't see Inkscape in the start menu (except if you are using the admin account that ran choco), so users will complain that "Inkscape is missing" and admins will have to use something else than choco.

If you (or whoever is to decide) say no, I nevertheless accept the decision. Then please close this PR.

**Comment by AdmiringWorm** [comment]
@pauby I am with @mgmax on this. Changing the passed in silent arguments to explicitly specify admin/all user mode to be used very rarely causes problems.

The only time I have seen this cause any problems in the past is if the developers of the installer has decided to use hard-coded values (usually related to installation Paths).

In general, we should probably start enforcing all CMAKE/WiX installers to always specify the `ALLUSERS=1` parameter for consistency’s sake (but that is a different problem).

We can have a discussion regarding this next week when everyone is back if you'd like.

**Comment by pauby** [comment]
I'm just coming back to this after the previous comment notification.

> The workaround does no harm even after the fix has landed. Is this one year worth the price of a workaround here?

If that is the timescale for the author to fix it then I'd agree we should fix it here.

**Review by AdmiringWorm — APPROVED** [review]
LGTM

**Comment by AdmiringWorm** [comment]
@mgmax your changes have been merged. Thanks for your contribution 👍

I do apologize for the long delay in getting this merged. I haven't pushed up a fix release for inkscape, as such this change won't go out until the software has a new version available.

**Comment by mgmax** [comment]
Thanks for merging. Coincidentally, there is already a newer Inkscape version available, which is an important bugfix release. I created a separate issue for that:
https://github.com/chocolatey-community/chocolatey-packages/issues/2392

**Pull request merged**
